### PR TITLE
Added external script for fed_job API

### PR DESCRIPTION
**Pull request by yhwen**
Fixes # .

### Description

1. Fixed an issue fed_job missing the attribute with "_" in front of parameter.
2. Add external script support for the Fed_job API.
3. 

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Quick tests passed locally by running `./runtest.sh`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated.

### CODE REVIEW

File: nvflare/job_config/fed_job.py [hunk 1]
```diff
@@ -161,6 +161,10 @@ def to(
                 else:
                     print(f"{target} already set to use GPU {self._gpus[target]}. Ignoring gpu={gpu}.")
             self._deploy_map[target].add_executor(obj, tasks=tasks)
+        elif isinstance(obj, str):  # treat the str type object as external script
+            if target not in self._deploy_map:
+                self._deploy_map[target] = ExecutorApp()
```
[the remark this answers is not in the document]
Comment (yhwen) [reply nvflare/job_config/fed_job.py:166]: FedJob.to() does not have the deploy sequence order. Not sure what kind of proper validation check can be properly done here. The way FedJob.to() is designed to deploy each component one by one. Maybe in the future we can change to a one time "export()" at the end, then we can add in the validation process before the export.

[the remark this answers is not in the document]
Comment (holgerroth) [reply nvflare/job_config/fed_job.py:166]: The ValueError linked above is already enforcing the order of running the to() command though. The point is that scripts could be sent to the server or clients when calling to(). So, we should make sure there's already a controller/executer defined before sending scripts with to().

[the remark this answers is not in the document]
Comment (holgerroth) [reply nvflare/job_config/fed_job.py:166]: Yes, this error will be given if we put `isinstance(obj, str)` as a check inside the existing else statement.

**Review by YuanTingHsieh — APPROVED** [review]
thanks, LGTM

**Pull request merged**